### PR TITLE
Api to get all configurations

### PR DESCRIPTION
**Pull request by nikhilcr939**
This PR fetch's all the configurations configured for the subscription.

Changes made in this pr
- EmailTemplate class changed to TemplateInformation
- Added getall configuration
- Updated nuget package to 1.0.1
- Updated Internal client with GetAll configuration and Delete template

### CODE REVIEW

File: src/Kmd.Logic.Email.Client/Models/SendEmailRequest.cs [hunk 1]
```diff
@@ -28,14 +28,10 @@ public SendEmailRequest()
         /// <param name="providerConfigurationId">A unique identifier that
         /// represents the associated configuration
         /// which this Email message will be sent with.</param>
-        /// <param name="subject">Email Subject</param>
```
Comment (avin101) [inline src/Kmd.Logic.Email.Client/Models/SendEmailRequest.cs]: Why are `subject` and `templates` removed?

File: src/Kmd.Logic.Email.Client/IInternalClient.cs [hunk 2]
```diff
@@ -164,7 +178,7 @@ internal partial interface IInternalClient : System.IDisposable
         /// <param name='cancellationToken'>
         /// The cancellation token.
         /// </param>
-        Task<HttpOperationResponse<IList<EmailTemplate>>> GetTemplateWithHttpMessagesAsync(System.Guid subscriptionId, System.Guid providerConfigurationId, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
+        Task<HttpOperationResponse<IList<TemplateInformation>>> GetTemplateWithHttpMessagesAsync(System.Guid subscriptionId, System.Guid providerConfigurationId, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
```
Comment (smahapatra86) [inline src/Kmd.Logic.Email.Client/IInternalClient.cs:181]: Maybe you can add to pr description about what all have been changed

**Comment by avinashbr123** [comment]
Needs to bump Nuget package version

File: CHANGES.md [hunk 3]
```diff
@@ -1,3 +1,6 @@
+1.0.1
+* Added get all configurations
```
Comment (smahapatra86) [inline CHANGES.md:2]: I think it also has template related `Delete` feature which was not there earlier. You can mention those as well here.

**Review by smahapatra86 — APPROVED** [review]

**Pull request merged**